public/app/features/variables/inspect/NetworkGraph.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { FC, useCallback, useEffect } from 'react';
+// @ts-ignore
+import vis from 'visjs-network';
+import { GraphEdge, GraphNode, toVisNetworkEdges, toVisNetworkNodes } from './utils';
+
+interface OwnProps {
+  nodes: GraphNode[];
+  edges: GraphEdge[];
+  direction?: 'UD' | 'DU' | 'LR' | 'RL';
+  onDoubleClick?: (node: string) => void;
+  width?: string;
+  height?: string;
+}
+
+interface ConnectedProps {}
+
+interface DispatchProps {}
+
+type Props = OwnProps & ConnectedProps & DispatchProps;
+
+export const NetWorkGraph: FC<Props> = ({ nodes, edges, direction, width, height, onDoubleClick }) => {
+  let network: any = null;
+  let ref: any = null;
```

Comment:
nit: instead of using the `ref` like this you can use the [useRef](https://reactjs.org/docs/hooks-reference.html#useref) hooks.

Reply:
great findings!!